### DISCUSSION TITLE
Can brick support a sixel image widget?

### DISCUSSION THREAD

**Discussion by amano-kenji** [jtdaugherty/brick#427] [Q&A] [answered]
Sixel is a terminal image display protocol. It has its own terminal escape sequences.

Go's tcell terminal rendering library doesn't support sixel. Someone submitted a pull request that adds sixel support to tcell, but it hasn't been merged for nearly 2 years. I expect it to take at least a decade for it to be merged if it is merged at all. Pull requests that aren't merged for months tend to be ignored for ever. I love brick because its maintainer is quick to merge or close pull requests.

I wonder whether brick can be made to support a sixel image widget. Can it support sixel images?

**Comment by jtdaugherty** [comment, marked as answer]
This is more of a `vty` question; `brick` could support sixel images so long as they were an `Image` primitive in `vty`. If someone wanted to work on a patch for `vty` to add them, they would Just Work in `brick` via [`Brick.Widgets.Core.raw`](https://hackage.haskell.org/package/brick-1.5/docs/Brick-Widgets-Core.html#v:raw). (I say "someone" since I'm not inclined to do it myself, but I'm not opposed to the feature.)

**Reply by amano-kenji** [reply]
It's good to know that brick and vty can at least be made to support sixel as a widget.
